frontend/src/app/App.tsx

```diff
@@ -109,9 +110,26 @@ const App = () => {
 
   if (error) {
     if (appInsights instanceof ApplicationInsights) {
+      let decoded: any;
+      let validToken = false;
+      if (accessToken) {
+        try {
+          decoded = jwtDecode(accessToken);
+          validToken = true;
+        } catch (e) {
+          validToken = false;
+          console.error("Failed to decode access token", e);
+        }
+      }
+      const rolesFieldName = `${process.env.REACT_APP_OKTA_TOKEN_ROLE_CLAIM}`;
       appInsights.trackException({
         exception: error,
-        properties: { "user message": "Server connection error" },
+        properties: {
+          "user message": "Server connection error",
+          validToken,
```

Comment:
minor nitpicks about this field,
- not sure if we really need it, since the other two fields wont be there when it says false, so it doesnt add any value
- it not named correctly, since it just means the `access_token` is set , not necessary `valid`
- if we were to keep it, I would give it a the key to something `"access token set"` that way it follows the other key naming scheme of this object, sticks with its camel casing.

Reply:
It is possible for the access token to be set but invalid, so not sure I agree that the field is unnecessary. Setting it to `false` initially does mean that it'll be hard to tell if the issue is that the access token hasn't been set at all or if the token couldn't be decoded successfully though. IMO it'd be better to remove the `false` initialization and give the field a key of something like `valid access token`. That way if the access token wasn't set at all, we'll get a `null` value in the logs, and otherwise it will tell us whether or not the token was valid.

Reply:
yeah I was iffy on that field but I was thinking it could be possible that the other fields would not be defined while the token was still valid, say if it just doesn't contain the data we're expecting, maybe due to some misconfig with the custom Okta claims. but missing and invalid tokens will look the same in the logs so I'll fix that and change the key to match the style of our other ones